core/server/list.go

```diff
@@ -0,0 +1,50 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/weaveworks/weave-gitops/core/clustersmngr"
+)
+
+// listFn is a function that contains the logic how to list objects
+// from a single namespace.
+type listFn func(context.Context, clustersmngr.Client, string) ([]interface{}, error)
+
+// listObejcts lists objects across all namespaces.
+//
+// This function can be simplified with generics once we move to Go 1.18,
+// but for now, it's a weird interface{} magic.
+func (cs *coreServer) listObjects(
+	ctx context.Context,
+	namespace string,
+	list listFn,
+) ([]interface{}, error) {
+	clustersClient := clustersmngr.ClientFromCtx(ctx)
+
+	if namespace != "" {
+		res, err := list(ctx, clustersClient, namespace)
+
+		return res, err
+	}
+
+	var results []interface{}
+
+	nsList, found := cs.cacheContainer.Namespaces()[clustersmngr.DefaultCluster]
```

Comment:
I think we might want to handle this differently. After we sort out the namespace thing, we need to make sure this implementation calls all clusters in that list because even tho this code live in the core, the way enterprise integrate it is by hooking the `coreServer` in tho their server but with a different clusters fetcher. If we "hardcode" the cluster name here that will be broken once they start using it.